libvast/vast/fbs/type.fbs

```diff
@@ -0,0 +1,106 @@
+namespace vast.fbs;
+
+/// A boolean value that can either be true or false.
+table BoolType {}
+
+/// A signed integer.
+table IntType {}
+
+/// An unsigned integer.
+table CountType {}
+
+/// A floating-point value.
+table RealType {}
+
+/// A time interval.
+table DurationType {}
+
+/// A point in time.
+table TimeType {}
+
+/// A string of characters.
+table StringType {}
+
+/// A regular expression.
+table PatternType {}
+
+/// An IP address (v4 or v6).
+table AddressType {}
+
+/// A CIDR subnet.
+table SubnetType {}
+
+/// A transport-layer port.
+table PortType {}
+
+/// An enumeration type that can have one specific value.
+table EnumerationType {
+  fields: [string];
```

Comment:
Could you change this to use an `EnumerationField` with a definition reflecting `RecordField`:
```
table EnumerationField {
  name: string;
}
```

I request this because I intend to add a type member in the not too far away future, and making this forwards compatible would be great!

Reply:
Sure, adding this table is possible.

It'd be still great to understand the underlying use case that motivates this. Mind explaining this real quick?

Reply:
I intend to let `enumeration`s grow into a real sum type instead of adding a separate type to the variant.

Reply:
I would rather have a separate `SumType` for this.

Reply:
Okay, we resolved this offline. The misunderstanding was that we had different notions of the power of `enum`s. My view was way more limited than, say, Rust or Haskell enums, which are effectively special cases of sum types.